lib/trino-parquet/src/test/java/io/trino/parquet/reader/TestParquetReader.java

```diff
@@ -98,4 +107,33 @@ public void testColumnReaderMemoryUsage()
         reader.close();
         assertThat(memoryContext.getBytes()).isEqualTo(0);
     }
+
+    @Test
+    public void testEmptyRowRangesWithColumnIndex()
+            throws URISyntaxException, IOException
+    {
+        List<String> columnNames = ImmutableList.of("l_shipdate", "l_commitdate");
+        List<Type> types = ImmutableList.of(DATE, DATE);
+
+        ParquetDataSource dataSource = new FileParquetDataSource(
+                new File(Resources.getResource("lineitem_sorted_by_shipdate.parquet").toURI()),
+                new ParquetReaderOptions());
+        ParquetMetadata parquetMetadata = MetadataReader.readFooter(dataSource, Optional.empty());
+        assertThat(parquetMetadata.getBlocks().size()).isEqualTo(2);
+        // The predicate and the file are prepared so that page indexes will result in non-overlapping row ranges and eliminate the entire first row group
```

Comment:
It would be worth to share `parquet-tools meta` relevant output

```
l_shipdate:       INT32 GZIP DO:505023 FPO:507443 SZ:7505/12562/1.67 VC:41083 ENC:BIT_PACKED,RLE,PLAIN_DICTIONARY ST:[min: 1992-01-04, max: 1996-08-31, num_nulls: 0]
l_commitdate:     INT32 GZIP DO:512528 FPO:515062 SZ:55183/60099/1.09 VC:41083 ENC:BIT_PACKED,RLE,PLAIN_DICTIONARY ST:[min: 1992-02-02, max: 1996-11-17, num_nulls: 0]


l_shipdate:       INT32 GZIP DO:1292991 FPO:1294199 SZ:3779/6083/1.61 VC:19092 ENC:BIT_PACKED,RLE,PLAIN_DICTIONARY ST:[min: 1996-08-31, max: 1998-11-29, num_nulls: 0]
l_commitdate:     INT32 GZIP DO:1296770 FPO:1298102 SZ:23853/26216/1.10 VC:19092 ENC:BIT_PACKED,RLE,PLAIN_DICTIONARY ST:[min: 1996-06-10, max: 1998-10-28, num_nulls: 0]
```

Reply:
actually using assertions for `l_shipdate` & `l_commitdate` would be more descriptive.

Reply:
Row group statistics are not enough here, the values are selected with page level min/max also in mind. It would make the comment/code too verbose to reproduce/assert all the relevant statistics here, I'm going to keep this as is.